samples/subsys/ipc/openamp/remote/boards/mps2_an521_remote.overlay

```diff
@@ -0,0 +1,20 @@
+/*
```

Comment:
nothing wrong on this line, but the commit message confuses me a bit.

> with sysbuild, kconfig overlays for boards cannot be passed to the remote
core. This means config file overlays must be present for remote cores as
well.

but here you are adding devicetree overlays ?
Secondly, sysbuild does support passing overlays to other builds (cores), like this:
`-D<image>_DTC_OVERLAY_FILE=<filename>`

but assume you are referring to automatically picking up board overlays for an image built as part of sysbuild.
Could you be a bit more precise on the message, and also feel free to comment if you think there is an extra use-case that sysbuild should support, or if you find this solution correct.

Reply:
I will update the commit message here to be correct, as the issue is that sysbuild does not allow automatically passing devicetree overlays between builds.

`-D<image>_DTC_OVERLAY_FILE=<filename>` does work correctly, but in this case the desired behavior is that the overlay CMake selects for the primary core is then automatically used for the secondary core. Since both projects are added as external projects, the `DTC_OVERLAY_FILE` variable for the main application cannot be accessed in `sysbuild.cmake`, and thus cannot be passed onto the build for the secondary core (except via commandline, as you mentioned)

Reply:
> the `DTC_OVERLAY_FILE` variable for the main application cannot be accessed in `sysbuild.cmake`, and thus cannot be passed onto the build for the secondary core

with `zephyr_get()` in place, this is actually possible.
https://github.com/zephyrproject-rtos/zephyr/pull/49674

we load the local sysbuild information for DTC_OVERLAY_FILE here:
https://github.com/zephyrproject-rtos/zephyr/blob/4a702ea21ef82b301464551807358f8490951e45/cmake/modules/configuration_files.cmake#L92

we could loosen up on this, so that an image may fetch the global overlay file, but we need to be really careful, cause in some cases images doesn't want to know about other images overlays.

So we should specify a clear pattern that is logical to users as to when such overlays are used be a single or multiple images.
Also we must consider that there can be multi images where two images wants the same overlay, but not the 3rd image.
(two cores where both want the overlay, and then a third app image on just one of the cores that doesn't want this overlay)